### PR TITLE
fix: pkg build scripts and post-release tagging

### PR DESCRIPTION
**Pull request by edwardfoyle**
#### Description of changes
A few fixes to our pkg build process:
1. building for all platforms in the dev pipeline
2. modify the `publish-to-verdaccio` script to publish with the release candidate pre-id
3. fix the `tag_rc.sh` script so it correctly pushes the release tag and the tags of all the other packages

#### Description of how you validated changes
Ran new steps locally and in CCI
#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,8 +33,8 @@
     "pkg-clean": "rimraf build out pkg/node_modules pkg/yarn.lock",
     "pkg-all": "source .circleci/local_publish_helpers.sh && generatePkgCli",
     "pkg-all-local": "yarn verdaccio-start && yarn verdaccio-connect && yarn publish-to-verdaccio && yarn pkg-all && yarn verdaccio-disconnect",
-    "publish:dev": "lerna publish --preid=rc.$(git rev-parse --short HEAD) --pre-dist-tag rc --exact --include-merged-tags --no-verify-access --conventional-prerelease --conventional-commits --yes --no-push",
-    "publish:rc": "lerna publish --preid=rc.$(git rev-parse --short HEAD) --pre-dist-tag rc --exact --include-merged-tags --no-verify-access --conventional-prerelease --conventional-commits --yes --no-push",
```
Comment (edwardfoyle) [inline package.json:37]: @danielleadams did you intend to specify `--include-merged-tags` here? Based on the [docs](https://github.com/lerna/lerna/tree/main/commands/version#--include-merged-tags) it doesn't seem like we need it but perhaps we do?

Comment (danielleadams) [reply package.json:37]: I don't think we need it. When i copied to script over from `master`, it was also using the flag, so I left it https://github.com/aws-amplify/amplify-cli/pull/10703

File: scripts/tag-rc.sh [hunk 2]
```diff
@@ -7,5 +7,9 @@ fi
 
 rc_tag="v$(jq -r .version packages/amplify-cli/package.json)"
 commit="$(git rev-parse HEAD~1)"
+short_commit="$(git rev-parse --short HEAD~1)"
 git tag $rc_tag $commit
-git push $rc_tag
+# push the main release tag
+git push origin $rc_tag
+# push the release tags for all other packages
+git tag | grep $short_commit | xargs git push origin
```
Comment (danielleadams) [inline scripts/tag-rc.sh:15]: Do we need to push the git tag for all the other packages?

Comment (edwardfoyle) [reply scripts/tag-rc.sh:15]: for now yes so that the release tags line up with what's published to npm

**Pull request merged**